lib/Runtime/Library/JavascriptFunction.cpp

```diff
@@ -2467,7 +2467,10 @@ void __cdecl _alloca_probe_16()
             this->functionInfo->IsAsync() ||
             this->IsGeneratorFunction() ||
             this->IsBoundFunction() ||
-            this->IsStrictMode()
+            this->IsStrictMode() ||
```

Comment:
Looks like this should also check to see if the function is a method, right? Seems like https://tc39.github.io/ecma262/#sec-forbidden-extensions implies this. This is a little complicated, though, because `FunctionInfo` doesn't have an `IsMethod` helper. To plumb the flag, you could pull it out of `PnFnc::IsMethod` in `ByteCodeGenerator::StartBindFunction`.

Reply:
I think `this->functionInfo->IsClassMethod()` up at the top of this clause captures that.

Reply:
I don't think that's true since `kFunctionIsClassMember` and `kFunctionIsMethod` are different flags on `PnFnc`.

Reply:
I see. In any case the behavior seems to be correct (per comparison other engines on test/es6/ES6RestrictedProperties.js). Will look into this in the future.

---
In reply to: [166775893](https://github.com/Microsoft/ChakraCore/pull/4651#discussion_r166775893) [](ancestors = 166775893)